### ISSUE TITLE
Explicit eta-expansion should not error (2.13) / warn (2.12)

### ISSUE THREAD

**Issue by Atry** [scala/bug#10612]
In current 2.13.x

```
scala> def foo() = 1
foo: ()Int

scala> def bar(f: () => Int) = {}
bar: (f: () => Int)Unit

scala> bar(foo _)

scala> foo _
       ^
       error: type mismatch;
        found   : Int
        required: () => ?
```

This is not intended according to @adriaanm's comment here https://github.com/scala/scala/pull/6206#discussion_r154388781

In 2.12.x, there's a corresponding warning

``` scala
scala> def foo() = 1
foo: ()Int

scala> def bar(f: () => Int) = {}
bar: (f: () => Int)Unit

scala> bar(foo _)

scala> foo _
<console>:13: warning: Eta-expansion of zero-argument method values is deprecated. Did you intend to write foo()?
       foo _
       ^
```

**Comment by Atry** [comment]
It's a regression because `foo _` does not cause the warning in Scala 2.12.3

**Comment by sjrd** [comment]
I thought this was as-designed, was it not?

**Comment by som-snytt** [comment]
There's some discussion at https://github.com/scala/scala/pull/5327 where Adriaan seeks to distinguish eta expansion and method values, and also extinguish method values.

**Comment by lrytz** [comment]
@adriaanm I updated the description to inlcude what we discussed [here](https://github.com/scala/scala/pull/6206#discussion_r154388781) and assigned this issue to you.

**Comment by som-snytt** [comment]
@adriaanm could comment on my fix from 22 days ago linked above https://github.com/scala/scala/pull/6177

**Comment by lrytz** [comment]
Thanks @som-snytt, I missed that!

**Comment by SethTisue** [comment]
@som-snytt just to confirm, the fix goes in 2.12.5, but when merged forward will fix the regression in 2.13...?
